### ISSUE TITLE
simsiam.py hardcodes bf16 autocast — the exact bug #39 fixed in main.py

### ISSUE THREAD

**Issue by lorenzoliuzzo** [lorenzoliuzzo/supervised-learning-on-food-images#43]
`src/simsiam.py:229`:

```python
with torch.autocast(device.type, dtype=torch.bfloat16, enabled=device.type == 'cuda'):
```

`select_amp_dtype` (`src/main.py:206`) exists precisely because bf16 on a pre-Ampere card -- Colab's free-tier Tesla T4, cc 7.5 -- does not error. It runs unaccelerated, measured at 12x slower (#39, c1a5656).

The fix landed in `main.py` only. `simsiam.py` was scaffolded in #38 (14cf407), the commit immediately before, and kept the hardcoded dtype.

This matters more here than it did in `main.py`: SimSiam is the **200-epoch** job (`--epochs` default 200) and the one most likely to be run on Colab rather than the local box.

### Fix

`simsiam.py` already imports from `main`, so:

```python
from main import FoodX251Dataset, dataset_paths, save_checkpoint, select_amp_dtype
```

then pick the dtype once in `main()` and thread it into `train_one_epoch`. fp16 also needs the `GradScaler` that `main.py` carries -- `train_one_epoch` currently calls `loss.backward()` / `optimizer.step()` directly (`src/simsiam.py:233-235`), which silently underflows in fp16 without scaling.

### Also in this file

`--seed` (`src/simsiam.py:157`) only calls `torch.manual_seed`. The augmentation pipeline draws from `random` too, so a seeded SSL run is not reproducible. Compare `main.py:330`, which seeds both. Related: #33.
